src/rtctools_heat_network/heat_physics_mixin.py

```diff
@@ -1462,6 +1538,274 @@ def __pipe_heat_to_discharge_path_constraints(self, ensemble_member):
+                0.0,
+                np.inf
+            ))
+            #ensures it does not select the lowest temperature, but the closest temperature
+            #supplytemperature needs to be reducing in T
+            # TODO: this should use ordering strategy
+            for temperature in supply_temperatures[:-1]:
+                temp_selected = self.state(f"{b}__temperature_disc_{temperature}")
+                next_temp = supply_temperatures[supply_temperatures.index(temperature)+1]
+                constraints.append((
+                    ates_temperature - ates_temperature_disc - temp_selected*(temperature-next_temp),
+                    -np.inf,
+                    0.0
+                ))
+            """
+            This function adds constraints to ensure that only one temperature level is active for 
+            the ates temperature. Furthermore, it sets the
+            temperature variable to the temperature associated with the temperature of the integer
+            variable.
+            """
+            sum = 0.0
+            for temperature in supply_temperatures:
+                temp_selected = self.state(f"{b}__temperature_disc_{temperature}")
+                sum += temp_selected
+                big_m = max(supply_temperatures)
+                # Constraints for setting the temperature variable to the chosen temperature
+                constraints.append(
+                    ((temperature - ates_temperature_disc + (1.0 - temp_selected) * big_m), 0.0,
+                     np.inf)
+                )
+                constraints.append(
+                    ((temperature - ates_temperature_disc - (1.0 - temp_selected) * big_m), -np.inf,
+                     0.0)
+                )
+            if len(supply_temperatures) > 0:
+                # Constraint to ensure that one single temperature is chosen for every timestep
+                constraints.append((sum, 1.0, 1.0))
+
+            # dt change due to charging should be 0 when discharging. Because minimum bound is 0, ates_dt_charging can not be negative.
+            ates_dt_charging = self.state(f"{b}.Temperature_change_charging")
+            ates_dt_charging_nominal = self.variable_nominal(f"{b}.Temperature_change_charging")
+            big_m = bounds[f"{b}.Temperature_change_charging"][1]
+            # ensures no ates temperature change because of charging when discarging
+            constraints.append(((ates_dt_charging - is_buffer_charging * big_m)/ates_dt_charging_nominal, -np.inf, 0.0))
+
+            # #TODO temporary: still add relation to bound ates_dt_charging, should also be piecewise linear as a function of the stored heat, added heat and temperature difference
+            #maybe relation of tempeature_ates at first and last time step does not have to be the same for feasibility, maybe only the discrete variable.
+            constraints.append(((ates_dt_charging - .1 * heat_ates / heat_nominal - (
```

Comment:
In my opinion the main reason to have the ATES temperature state is to have a better approximation of the operational cost during discharging in combination with the HP. I therefore think that at least for now we shouldn't invest in a stored_heat depending ates_dt_charging